app/controllers/urls_controller.rb

```diff
@@ -5,28 +5,20 @@ def new
   end
 
   def create
-    @url = Url.new(url_params)
-    @url.clean_url
+    @url = CreateShortUrlService.new(url_params).call
 
-    if @url.new_link?
-      if @url.save
-        render js: "document.getElementById('short_url_box').classList
-          .remove('d-none');document.getElementById('short_url')
-          .innerText='http://#{request.host}/#{@url.url_short}/';"
-      else
-        flash[:alert] = @url.errors.full_messages
-        redirect_to :root
-      end
-    else
-      @url_in_database = Url.find_by_url_clean(@url.url_clean)
+    if @url.save
       render js: "document.getElementById('short_url_box').classList
-      .remove('d-none');document.getElementById('short_url')
-      .innerText='http://#{request.host}/#{@url_in_database.url_short}/';"
+        .remove('d-none');document.getElementById('short_url')
+        .innerText='http://#{request.host}/#{@url.url_short}/';"
+    else
+      flash[:alert] = @url.errors.full_messages
+      redirect_to :root
     end
   end
 
   def show
-    @url = Url.find_by_url_short(params[:url_short])
+    @url = Url.find_by(url_short: params[:url_short])
     redirect_to "http://#{@url.url_clean}/"
```

Comment:
What if `@url` is `nil`?
You can either use `find_by!`(notice the `!`) in combination with `rescue_from ActiveRecord::RecordNotFound` or manually handle it by checking whether the `@url` is present beforehand